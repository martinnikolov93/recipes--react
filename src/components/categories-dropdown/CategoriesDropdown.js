import React from 'react';
import styles from './CategoriesDropdown.module.css'
import { NavLink } from 'react-router-dom';

class CategoriesDropdown extends React.Component {
    constructor(props) {
        super(props)

        this.state = {
            categories: [],
            isHovering: false
        }
    }

    componentDidMount() {
        fetch('http://localhost:9999/api/category/')
            .then((response) => response.json())
            .then((categories) => {
                this.setState({ categories })
            })
            .catch((err) => console.log(err))
    }

    handleMouseEnter = () => {
        this.setState({ isHovering: true })
    }

    handleMouseLeave = () => {
        this.setState({ isHovering: false })
    }

    render() {
        return (
            <>
                <span

                >
                    <NavLink to='/categories' activeClassName={styles['nav-button-active']}
                        className={this.props.className}
                        onMouseEnter={this.handleMouseEnter}
                        onMouseLeave={this.handleMouseLeave}>
                        Categories
                    </NavLink>
                </span>
                {
                    this.state.isHovering &&
                    <span onMouseEnter={this.handleMouseEnter} onMouseLeave={this.handleMouseLeave} className={styles['categories-box']}>
                        {this.state.categories.map((category, i) => {
                            return (
                                <div key={i}>
                                    {i === 0 ? null : <hr />}
                                    <NavLink className={styles['category-link']} to={`/categories/${category.title}`} activeClassName={styles['nav-button-active']}>
                                        {category.title}
                                    </NavLink>
                                </div>
                            )
                        })}
                    </span>
                }
            </>
        )
    }
}

export default CategoriesDropdown