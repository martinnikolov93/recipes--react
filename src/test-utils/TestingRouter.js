import React from 'react';
import UserContext from '../Context';
import { BrowserRouter, Route } from 'react-router-dom'

const TestingRouter = ({user, children}) => {
    return (
        <BrowserRouter>
            <Route>
                <UserContext.Provider value={user}>
                    {children}
                </UserContext.Provider>
            </Route>
        </BrowserRouter>
    )
}

export default TestingRouter