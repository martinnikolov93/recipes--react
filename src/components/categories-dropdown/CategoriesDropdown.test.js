import React from 'react';
import renderer from 'react-test-renderer'
import CategoriesDropdown from './CategoriesDropdown';
import TestingRouter from '../../test-utils/TestingRouter';

const categoriesDummy = [
    { title: 'category1' },
    { title: 'category2' },
    { title: 'category3' },
    { title: 'category4' },
]

describe('Categories dropdown component', () => {
    it('Should render Categories dropdown component', () => {
        const tree = renderer.create(
            <TestingRouter>
                <CategoriesDropdown data={categoriesDummy} />
            </TestingRouter>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})