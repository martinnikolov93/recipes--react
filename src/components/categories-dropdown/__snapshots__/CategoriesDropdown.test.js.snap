// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories dropdown component Should render Categories dropdown component 1`] = `
<span>
  <a
    aria-current={null}
    href="/categories"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Categories
  </a>
</span>
`;
